audio/src/passthrough_decoder.rs

```diff
@@ -82,70 +39,126 @@ impl<R: Read + Seek> PassthroughDecoder<R> {
     /// Constructs a new Decoder from a given implementation of `Read + Seek`.
     pub fn new(rdr: R) -> Result<Self, PassthroughError> {
         let mut rdr = PacketReader::new(rdr);
-        let mut wtr = PacketWriter::new(Vec::new());
+        let stream_serial = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_millis() as u32;
 
-        let stream_serial = write_headers(&mut rdr, &mut wtr)?;
         info!("Starting passthrough track with serial {}", stream_serial);
 
+        // search for ident, comment, setup
+        let ident = get_header(1, &mut rdr)?;
+        let comment = get_header(3, &mut rdr)?;
+        let setup = get_header(5, &mut rdr)?;
+
+        // remove un-needed packets
+        rdr.delete_unread_packets();
+
         return Ok(PassthroughDecoder {
             rdr,
-            wtr,
-            lastgp_page: Some(0),
-            absgp_page: 0,
+            wtr: PacketWriter::new(Vec::new()),
+            ofsgp_page: 0,
             stream_serial,
+            ident,
+            comment,
+            setup,
+            eos: false,
+            bos: false,
         });
     }
 }
 
 impl<R: Read + Seek> AudioDecoder for PassthroughDecoder<R> {
     fn seek(&mut self, ms: i64) -> Result<(), AudioError> {
         info!("Seeking to {}", ms);
-        self.lastgp_page = match ms {
-            0 => Some(0),
-            _ => None,
-        };
+
+        // add an eos to previous stream if missing
+        if self.bos == true && self.eos == false {
```

Comment:
```suggestion
        if self.bos && !self.eos {
```
And below as well

Reply:
That I much prefer but I was under the impression that Rust always prefers you to explicitly write against what you are checking... (I would never do that == true in C for example...)